src/kafka-file/kafkaFileClient.ts

```diff
@@ -5,12 +5,114 @@ import { ClusterSettings } from "../settings/clusters";
 
 import { getLanguageModelCache, LanguageModelCache } from './languageModelCache';
 import { KafkaFileDocument } from "./languageservice/parser/kafkaFileParser";
-import { ConsumerLaunchStateProvider, getLanguageService, LanguageService, ProducerLaunchStateProvider, SelectedClusterProvider, TopicDetail, TopicProvider } from "./languageservice/kafkaFileLanguageService";
+import { AutoCreateTopicResult, ConsumerLaunchStateProvider, getLanguageService, LanguageService, ProducerLaunchStateProvider, SelectedClusterProvider, TopicDetail, TopicProvider } from "./languageservice/kafkaFileLanguageService";
 import { runSafeAsync } from "./utils/runner";
-import { TopicItem } from "../explorer";
-import { KafkaModelProvider } from "../explorer/models/kafka";
 import { ThrottledDelayer } from "./utils/async";
 import { WorkspaceSettings } from "../settings";
+import { BrokerConfigs } from "../client/config";
+import { KafkaModelProvider } from "../explorer/models/kafka";
+import { ClusterItem } from "../explorer/models/cluster";
+class ClusterInfo {
+
+    private topics?: TopicDetail[];
+
+    private autoCreateTopicEnabled?: AutoCreateTopicResult;
+
+    constructor(public readonly cluster?: ClusterItem, public readonly error?: any) {
+    }
+
+    async getTopics(): Promise<TopicDetail[]> {
+        if (this.topics) {
+            return this.topics;
+        }
+        try {
+            this.topics = await this.loadTopics();
+        }
+        catch (e) {
+            this.topics = [];
+        }
+        return this.topics;
+    }
+
+    private async loadTopics(): Promise<TopicDetail[]> {
+        if (this.cluster) {
+            return (await this.cluster.getTopics())
+                .map(child => (child.topic));
+        }
+        return [];
+    }
+
+    async getTopic(topicId: string): Promise<TopicDetail | undefined> {
+        const topics = await this.getTopics();
+        return topics.find(topic => topic.id === topicId);
+    }
+
+    async getAutoCreateTopicEnabled(): Promise<AutoCreateTopicResult> {
+        if (this.autoCreateTopicEnabled !== undefined) {
+            return this.autoCreateTopicEnabled;
+        }
+        try {
+            const result = await this.loadAutoCreateTopicEnabled();
+            this.autoCreateTopicEnabled = { type: result ? "enabled" : "disabled" };

```

Comment:
doesn't work for red panda, since there's no such available configuration, it should be "unknown"
<img width="1351" alt="Screenshot 2021-04-28 at 11 49 45" src="https://user-images.githubusercontent.com/148698/116384141-d9feca00-a817-11eb-8877-c481784e4be0.png">

Reply:
Yet Red Panda supports auto topic creation